packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js

```diff
@@ -423,111 +425,142 @@ function useMediaUploadQueue() {
     [startUploading, cancelUploading, uploadImage, uploadVideo]
   );
 
-  // Upload files to server, optionally first transcoding them.
-  useEffect(() => {
-    (async () => {
-      await Promise.all(
-        /**
-         * Uploads a single pending item.
-         *
-         * @param {Object} item Queue item.
-         * @param {File} item.file File object.
-         * @return {Promise<void>}
-         */
-        state.queue.map(async (item) => {
-          const {
-            id,
-            file,
-            state: itemState,
-            resource,
-            additionalData = {},
-            muteVideo,
-            trimData,
-          } = item;
-          if (ITEM_STATUS.PENDING !== itemState) {
+  const uploadWithTranscoding = useCallback(
+    async (item) => {
+      const { id, file, resource, muteVideo, trimData } = item;
+
+      const isVideo = file.type.startsWith('video/');
+
+      if (isVideo) {
+        const fileInfo = await getFileInfo(file);
```

Comment:
We have data for duration, height and width here? Could we use this data to popular the resource?

Reply:
Good question! I'd suspect we already have correct height and width here, but especially leveraging more accurate duration information makes sense. I'll add it.

Reply:
So it's not as easy as I thought, as it would likely require adding a new reducer to update an item's resource's metadata like that. Seems best suited for its own ticket to explore.